### ISSUE TITLE
Clarify relationship of CMAF, DASH, DASH IF and HLS mp4 media segments

### ISSUE THREAD

**Issue by johnsim** [cta-wave/content-specification-task-force#25]
(2018-09-12) What is the relationship between CMAF, DASH, DASH IF and HLS constraints on media segments? Are all CMAF-compliant media segments also DASH-compliant media segments? DASH-IF compliant media segments? HLS-compliant media segments?

**Comment by KilroyHughes** [comment]
Yes. CMAF Fragments and Segments are a constrained subset that conforms to those other specs and deployed players, including "segments" defined in ISOBMFF. But, CMAF Tracks, Switching Sets, Fragments, and derived Addressable Media Objects are derived from ISOBMFF segments, CENC, NAL Video, etc. normative references directly, not from any of the specs mentioned. 

Note that the scope of CMAF, the 'cmfc' container brand, CMAF Media Profiles, and the specs mentioned all go beyond "segments", so the answer is more complicated if the intent of the question goes beyond comparing DASH segment and HLS chunks to CMAF Deliverable Media Objects (Chunks, Segments, and Track Files).  

The bottom line is the CMAF content with an appropriate manifest should play on players that support DASH, DASH IF, HLS, etc. and the particular Media Profiles, encryption, and DRM in a CMAF Presentation. The additional constraints improve functionality and interop, such as seamless switching, seamless splicing, seamless playback of dynamic live streams, various ad insertion methods, low latency live, etc.

**Comment by irajs** [comment]
suggest to add a note to the content spec on this topic, referencing appropriate HLS spec version and also MPEG and DASH-IF specs. Also clarify the one-way relationship (not all HLS or DASH content are CMAF/WAVE compliant).

**Comment by johnsim** [comment]
Agreed in Oct 10, 2018 conference call that @irajs will draft paragraph for section 3.3 and recommend which specs should be included in the informative reference section (MPEG DASH, IETF HLS, DASH IF, etc.)

**Comment by haudiobe** [comment]
The problem with the additional restrictions in CMAF right now is manifold:
1) none of the available test content for DASH-IF and HbbTV follows the constraint
2) deployed services do not follow these constraints including HLS, Netflix, SmoothStreaming and Youtube
3) Open source packaging tools do not use these constraints be default
4) It is not known that these constraints are needed for anything
Based on this MPEG decided to create a new Amendment that will create cmf2 (as a working assumption). However, this will question the media profile conformance. There will be hard decisions to be made. Today cmfc is mostly useless and more disturbing for convergence than it is helpful. The document should add some information on this context.

**Comment by johnsim** [comment]
Agreed in October 24, 2018 conference call - we will add a generic note to Amendment 1 of the Content Specification which refers to how the normatively referenced specifications (MPEG, etc.) can change. This will permit MPEG to complete their work regarding 'cmfc' and 'cmf2'. If 'cmfc' becomes less constrained, this would result in the minimum changes to our content spec - perhaps in the Spring release.

**Comment by KilroyHughes** [comment]
Regarding #4 "It is not known that these constraints are needed for anything". 

These Header constraints enable single initialization playback and define a single container format that works for both single initialization and multiple initialization, i.e. appending the Header of each Track immediately before Fragments of that Track are appended to an MSE buffer. 

In combination with a "Master Header" to initialize the required media pipeline for a WAVE Program, 'cmfc' with single initialization constraints allows continuous playback functionally equivalent to MPEG-2 Transport Streams used for around 75% of adaptive streaming and live contribution feeds today. M2TS is defined in DASH and MSE to be "self-initializing" and has no headers. 

The required media pipeline initialization for an entire M2TS Program is typically specified in the manifest. An M2TS stream containing ads and programs signals all parameter changes in each CVS or sample and doesn't require header processing during playback. Picture reordering, duration, presentation time, and decoding parameters are signaled in each sample (DTS/PTS timestamps) or coded video sequence (SPS/PPS NALs). 

To encode spliced M2TS in CMAF without single initialization constraints requires breaking the M2TS stream into a sequence of CMAF files with different Headers to signal any changes in Decoded Picture Buffer removal delay, frame rate, resolution, picture dependencies, or sample entry. It is much easier to continue using M2TS streams for delivery if CMAF doesn't have equivalent single initialization capability. 

DASH has primarily been used for file playback or single Period live (often requiring re-encoding of live streams to eliminate splices and Periods). Single initialization functionality in CMAF is important to make it functionally competitive with M2TS and enable convergence on a common content format for WAVE. 

I think grouping all single initialization constraints under one brand is the best solution, since MPEG has decided to have two container formats and brands. Brands like 'cmfc' and 'cmfs' (for "single initialization") would make sense. The Header constraints in question and other single initialization constraints in CMAF should only apply to one brand that identifies single initialization constraints, which can then be recognized and processed by optimized players by appending a Master Header once at the start (other players can process a Master Header every Track switch, but there will be no parameter changes in the Header). 

The requirement on players is to handle the V1 'trun' box and inband video parameters ('avc3' and 'hev1' sample formats), as well as edit lists to handle both brands.
